include/exadg/structure/spatial_discretization/operator.cpp

```diff
@@ -453,19 +453,20 @@ Operator<dim, Number>::setup_preconditioner()
   }
   else if(param.preconditioner == Preconditioner::AMG)
   {
+    AMGData amg_data = param.multigrid_data.coarse_problem.amg_data;
+#ifdef DEAL_II_WITH_TRILINOS
+    amg_data.ml_operator_type = MLOperatorType::Elasticity;
+#endif
```

Comment:
We have this code for elasticity multiple times, and multiple times for Laplace. My hope with the design described in other comments would be that we only have to implement this once for each operator.

Reply:
removed!